src/signature_accumulator.rs

```diff
@@ -208,7 +208,11 @@ mod tests {
             });
         });
 
-        assert_eq!(count, expected_msgs_count);
+        // we expect every message to accumulate twice - as we will accumulate 8 signatures, and 3
+        // are enough to construct a fully signed message; once we get the first 3 signatures, the
+        // message will accumulate, get removed from the accumulator, and the remaining 5 sigs are
+        // enough for it to accumulate again
+        assert_eq!(count, expected_msgs_count * 2);
```

Comment:
That does seem less than ideal, we probably should not resend message here, what do you think?

Reply:
In production this will be handled by the message filter. I'm not sure what we could do here for the code to remain reasonably clean, other than not add all the signatures to the accumulator, but only some part of them.